spotler/backend/plugin/src/main/kotlin/com/ritense/spotler/autoconfiguration/SpotlerConfiguration.kt

```diff
@@ -3,18 +3,27 @@ package com.ritense.spotler.autoconfiguration
 import com.ritense.plugin.service.PluginService
 import com.ritense.spotler.plugin.SpotlerPluginFactory
 import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean
+import org.springframework.boot.web.client.RestTemplateBuilder
 import org.springframework.context.annotation.Bean
 import org.springframework.context.annotation.Configuration
+import org.springframework.web.client.RestTemplate
 
 @Configuration
 class SpotlerConfiguration {
 
     @Bean
     @ConditionalOnMissingBean(SpotlerPluginFactory::class)
     fun spotlerPluginFactory(
-        pluginService: PluginService
+        pluginService: PluginService,
+        restTemplate: RestTemplate
     ) = SpotlerPluginFactory(
-        pluginService
+        pluginService,
+        restTemplate
     )
 
+    @Bean
+    @ConditionalOnMissingBean(RestTemplate::class)
+    fun restTemplate(): RestTemplate {
+        return RestTemplateBuilder().build()
```

Comment:
This is needs to have more that just a plain RestTemplate. Please include a timeout if the request takes to long.

Reply:
RestTemplateBuilder()
            .setConnectTimeout(Duration.ofSeconds(10))
            .setReadTimeout(Duration.ofSeconds(10))
            .build()